src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,211 @@
+        private readonly IDirectoryManager _directoryManager;
+        private readonly IFileManager _fileManager;
+        private readonly OrchestratorSession _session;
+        private readonly string _targetApplicationFullPath;
+        
+        public GenerateDeploymentProjectCommand(
+            IToolInteractiveService toolInteractiveService,
+            IConsoleUtilities consoleUtilities,
+            ICdkProjectHandler cdkProjectHandler,
+            ICommandLineWrapper commandLineWrapper,
+            IDirectoryManager directoryManager,
+            IFileManager fileManager,
+            OrchestratorSession session,
+            string targetApplicationFullPath)
+        {
+            _toolInteractiveService = toolInteractiveService;
+            _consoleUtilities = consoleUtilities;
+            _cdkProjectHandler = cdkProjectHandler;
+            _commandLineWrapper = commandLineWrapper;
+            _directoryManager = directoryManager;
+            _fileManager = fileManager;
+            _session = session;
+            _targetApplicationFullPath = targetApplicationFullPath;
+        }
+
+        /// <summary>
+        /// This method takes a user specified directory path and generates the CDK deployment project at this location.
+        /// If the provided directory path is an empty string, then a default directory is created to save the CDK deployment project.
+        /// </summary>
+        /// <param name="saveCdkDirectoryPath">An absolute or a relative path provided by the user.</param>
+        /// <param name="customRecipeName">The custom name of the recipe that is saved as part of the recipe definition</param>
+        /// <returns></returns>
+        public async Task ExecuteAsync(string saveCdkDirectoryPath, string customRecipeName)
+        {
+            var orchestrator = new Orchestrator(_session, new[] { RecipeLocator.FindRecipeDefinitionsPath() });
+            var recommendations = await GenerateDeploymentRecommendations(orchestrator);
+            var selectedRecommendation = _consoleUtilities.AskToChooseRecommendation(recommendations);
+
+            if (string.IsNullOrEmpty(saveCdkDirectoryPath))
+                saveCdkDirectoryPath = GenerateDefaultSaveDirectoryPath();
+
+            var newDirectoryCreated = CreateSaveCdkDirectory(saveCdkDirectoryPath);
+
+            var errorMessage = ValidateSaveCdkDirectory(saveCdkDirectoryPath);
+            if (!string.IsNullOrEmpty(errorMessage))
+            {
+                if (newDirectoryCreated)
+                    _directoryManager.Delete(saveCdkDirectoryPath);
+                throw new InvalidSaveDirectoryForCdkProject(errorMessage);
+            }
+            
+            var directoryUnderSourceControl = await IsDirectoryUnderSourceControl(saveCdkDirectoryPath);
+            if (!directoryUnderSourceControl)
+            {
+                var userPrompt = "The target directory for saving the CDK " +
```

Comment:
My suggest:
`Warning: The target directory is not being tracked by source control. If the saved saved deployment project is used for deployment it is important the deployment project is retained to allow future redeployments to deployed applications.`

Reply:
Fixed